modules/video_writer.py

```diff
@@ -93,14 +93,17 @@ def _init_video(self, image: np.ndarray):
         # )
 
 
-    def add_frame(self, image: np.ndarray):
+    def add_frame(self, image: np.ndarray, timestamp=None):
 
         # First frame initialization
         if self._video is None:
             self._init_video(image=image)
         
         if self._include_timestamp_overlay:
-            ts = self._get_timestamp_str()
+            if timestamp is not None:
```

Comment:
Modified the way timestamps are being added to allow for them to be added from given info.